api-spec/protobuf/ark/v1/indexer.proto

```diff
@@ -0,0 +1,222 @@
+    option (google.api.http) = {
+      get: "/v1/history/{address}"
+    };
+  };
+  rpc GetVtxoChain(GetVtxoChainRequest) returns (GetVtxoChainResponse) {
+    option (google.api.http) = {
+      get: "/v1/vtxo/{outpoint.txid}/{outpoint.vout}/chain"
+    };
+  }
+  rpc GetVirtualTxs(GetVirtualTxsRequest) returns (GetVirtualTxsResponse) {
+    option (google.api.http) = {
+      get: "/v1/virtualTx/{txids}"
+    };
+  }
+  rpc GetSweptCommitmentTx(GetSweptCommitmentTxRequest) returns (GetSweptCommitmentTxResponse) {
+    option (google.api.http) = {
+      get: "/v1/commitmentTx/{txid}/swept"
+    };
+  }
+}
+
+message GetCommitmentTxRequest {
+  string txid = 1;
+}
+message GetCommitmentTxResponse {
+  int64 started_at = 1;
+  int64 ended_at = 2;
+  map<uint32, IndexerBatch> batches = 3;
+}
+
+message GetVtxoTreeRequest {
+  IndexerOutpoint batch_outpoint = 1;
+  IndexerPageRequest page = 2;
+}
+message GetVtxoTreeResponse {
+  repeated IndexerNode vtxo_tree = 1;
+  IndexerPageResponse page = 2;
+}
+
+message GetForfeitTxsRequest {
+  IndexerOutpoint batch_outpoint = 1;
+  IndexerPageRequest page = 2;
+}
+message GetForfeitTxsResponse {
+  repeated string txs = 1;
+  IndexerPageResponse page = 2;
+}
+
+message GetConnectorsRequest {
+  IndexerOutpoint batch_outpoint = 1;
+  IndexerPageRequest page = 2;
+}
+message GetConnectorsResponse {
+  repeated IndexerNode connectors = 1;
+  IndexerPageResponse page = 2;
+}
+
+message GetSpendableVtxosRequest {
+  string address = 1;
+  IndexerPageRequest page = 2;
+}
+message GetSpendableVtxosResponse {
+  repeated IndexerVtxo vtxos = 1;
+  IndexerPageResponse page = 2;
+}
+
+message GetTransactionHistoryRequest {
+  string address = 1;
+  int64 start_time = 2;
+  int64 end_time = 3;
+  IndexerPageRequest page = 4;
+}
+message GetTransactionHistoryResponse {
+  repeated IndexerTxHistoryRecord history = 1;
+  IndexerPageResponse page = 2;
+}
+
+message GetVtxoChainRequest {
+  IndexerOutpoint outpoint = 1;
+  IndexerPageRequest page = 2;
+}
+message GetVtxoChainResponse {
+  map<string, IndexerTransactions> graph = 1;
+  IndexerPageResponse page = 2;
+}
+
+message GetVirtualTxsRequest {
+  repeated string txids = 1;
+  IndexerPageRequest page = 2;
+}
+message GetVirtualTxsResponse {
+  repeated string txs = 1;
+  IndexerPageResponse page = 2;
+}
+
+message GetSweptCommitmentTxRequest {
+  string txid = 1;
+}
+message GetSweptCommitmentTxResponse {
+  repeated string swept_by = 1;
+}
+
+message SubscribeForAddressesRequest {
+  repeated string addresses = 1;
+}
+message SubscribeForAddressesResponse {
+  string address = 1;
+  repeated IndexerVtxo new_vtxos = 2;
+  repeated IndexerVtxo spent_vtxos = 3;
+}
+
+message IndexerTransactions {
+  repeated IndexerChain txs = 1;
+  int64 expires_at = 2;
+}
+
+message IndexerBatch {
+  uint64 total_batch_amount = 1;
+  uint64 total_forfeit_amount = 2;
+  int32 total_input_vtxos = 3;
```

Comment:
I thought we were removing this type (inputs) of data from the indexer? I would also move them from being batch specific to CommitmentTx level

Reply:
`total_input_vtxos` here refers to the count of vtxos forfeited during the batch process

Reply:
i prefer redundancy here and have `total_input_vtxos` also in GetCommitmentTx as a cumulative sum

Reply:
Let's say a multi batch tx happens and:
a user has a VTXO of 0.1 BTC but wants to get 2 VTXOs of 0.5 each.

These 0.5 VTXOs potentially end up in 2 different batches. There is only one forfeit.  Only a total_input_vtxos  on GetCommitmentTx  would make sense. 

When a round is being done, intents should be on the round, batches only dictate vtxo and tree signing, but the forfeiture is on a commitment tx scope?